base/intfuncs.jl

```diff
@@ -1205,6 +1205,7 @@ Base.@assume_effects :terminates_locally function binomial(n::T, k::T) where T<:
     end
     copysign(x, sgn)
 end
+binomial(n::Integer, k::Integer) = binomial(promote(n, k)...)
```

Comment:
Nitpick: I would have moved this general definition up, to attach it to the docstring.

Reply:
Fixed in #56679